[plaster] Adding presubmit check for 🩹 files

cdesouza-chromium opened the pull request:
This change adds support to plaster for `.patchinfo` files, allowing a
more efficient way to make sure any given plaster file has should be
regenerated. This introduces the necessary supporting machinery to
permit a `plaster.py check` call, that is being hooked into our
presubmits.

This change aims to mitigate unwanted/accidental changes to patch files
for which a plaster file already exists.

### What does plaster do to .patchinfo files

Plaster now generates the contents of a .patchinfo file, and based on that content it determines if one of these files must be updated: source, patch, or patchinfo (a combination of `npm run update_patches` and `npm run apply_patches`). A new key is being added to the patchinfo format, namely, `plaster`, which is used to store the path and checksum for the plaster file for a given patch/source set. Files are only updated on disk if the resulting new content doesn't match the hash of the file saved in disk.

For now plaster always produce a .patchfile when processing a plaster, to avoid unnecessary writes and to confirm that submitted patches do have the expected content. This may change in the future with a more mature approach where the contents of the patchinfo file may be used to avoid running all the plaster transformations. This means that for now we have to pay for a full plaster generation to determine if a file source/patch/patchinfo are out-of-date.

Resolves https://github.com/brave/brave-browser/issues/45058

brave-builds commented:
> [!WARNING]
> You have got a presubmit warning. Please address it if possible.

```
Found 12 lines longer than 80 characters (first 5 shown).
```

Items:
```
PRESUBMIT.py, line 142, 83 chars
PRESUBMIT.py, line 143, 84 chars
PRESUBMIT.py, line 144, 93 chars
PRESUBMIT.py, line 151, 83 chars
PRESUBMIT.py, line 152, 84 chars
```

<!-- presubmit-hash=fa89105db8f85cbdfced97338e9774cc6d7751397f8a0c0b3ec17eadca1a90c7 -->

petemill commented:
> This change adds support to plaster for .patchinfo files, allowing a more efficient way to make sure any given plaster file has should be regenerated. 

@cdesouza-chromium would you mind explaining how it interacts with the .patchinfo file - for example does it just read the info (and then do what with it?) or does it also contribute to writing some data to the file, along with the existing patching JS class? Just a little summary of how the support for .patchingof files allows the effect you describe would be helpful.

cdesouza-chromium commented:
> > This change adds support to plaster for .patchinfo files, allowing a more efficient way to make sure any given plaster file has should be regenerated.
> 
> @cdesouza-chromium would you mind explaining how it interacts with the .patchinfo file - for example does it just read the info (and then do what with it?) or does it also contribute to writing some data to the file, along with the existing patching JS class? Just a little summary of how the support for .patchinfo files allows the effect you describe, would be helpful.

@petemill sorry for the long time I'm taking with this. I have updated the PR description, to describe what we are doing with `.patchinfo` files, but I also added extensive documentation to the `Patchinfo` class in `plaster.py` also explaining what it does.

cdesouza-chromium commented:
Im breaking this PR apart into multiple other PRs.
